### ISSUE TITLE
[Installer]: change authProviders from raw data to a secret

### ISSUE THREAD

**Issue by MrSimonEmms** [gitpod-io/gitpod#6867]
Currently, the `authProviders` in the config.yaml is an array of objects with sensitive data in. **THIS IS A BAD IDEA*** and will probably raise concern with self-hosted users with additional security/compliance requirements.

Current - array of objects

```yaml
authProviders:
  - id: Public-GitHub
    host: github.com
    type: GitHub
    oauth:
      clientId: xxx
      clientSecret: xxx
      callBackUrl: https://$DOMAIN/auth/github.com/callback
      settingsUrl: https://github.com/organizations/$ORG_ID/settings/applications/$APP_ID
```

Proposed - array of secrets

```yaml
authProviders:
  - kind: secret
    name: auth-provider-github
  - kind: secret
    name: auth-provider-gitlab
```

Each secret would then have the data in an expected format similar to the current auth provider config.

One of the difficulties with this is that this is injected into `server` as a [ConfigMap](https://github.com/gitpod-io/gitpod/blob/8884ea28db71304895234a791f09cb5827e554fb/installer/pkg/components/server/configmap.go#L53), so the object would need to be amended to accept a file location and then read the content from the file

**Comment by iQQBot** [comment]
I think all these things should be configurable in the Admin UI at a later stage, and you probably don't want to have to reinstall a global authProviders once

**Comment by MrSimonEmms** [comment]
@iQQBot there is a section in the UI for this already - you can add and remove config once set

**Comment by iQQBot** [comment]
Did you mean "Integrations" on "Setting"?
This is not the same as what I'm talking about, I'm talking about the kind that can be configured in the Admin UI for all users and can appear on the login screen

**Comment by csweichel** [comment]
/team meta

This would affect meta and how that configuration is consumed.

**Comment by AlexTugarev** [comment]
@iQQBot, it's the Setup [page](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/src/Setup.tsx) which is meant here. It allows you to setup an Git Integration right after deployment when no auth provider is provided by configuration. That used to be our self-hosted path, as it allows to set up OAuth more user friendly.

👍🏻 for converting those to secrets.

**Comment by iQQBot** [comment]
Oh, I forgot it

**Comment by MrSimonEmms** [comment]
Not sure if this would be applicable, but I did something very similar in `content-service-api` in #6777. I added an additional `AccessKeyIdFile` and `SecretAccessKeyFile` param to the config - if they're set, the constructor reads the contents from the specified file and sets to `AccessKeyID` and `SecretAccessKey`. I wonder if a similar approach could be followed here?

That way, it keeps the same functionality in the codebase except for the declaration of the secret.
